scripts/tests/test_zkai_attention_kv_quantized_softmax_receipt_gate.py

```
@@ -0,0 +1,136 @@
+        source["score_rows"][0]["weight_denominator"] = 0
+        with self.assertRaisesRegex(gate.QuantizedSoftmaxReceiptGateError, "source input validation drift|denominator"):
+            gate.validate_quantized_kernel(source)
+
+        source = copy.deepcopy(self.source)
+        source["score_rows"][0]["output_remainder"][0] = 999
+        with self.assertRaisesRegex(gate.QuantizedSoftmaxReceiptGateError, "source input validation drift|quotient|remainder"):
+            gate.validate_quantized_kernel(source)
+
+    def test_write_json_and_tsv_round_trip(self):
+        with tempfile.TemporaryDirectory() as tmp:
+            tmp = gate.pathlib.Path(tmp)
```

Ruff (0.15.12)

[warning] `with` statement variable `tmp` overwritten by assignment target

(PLW2901)